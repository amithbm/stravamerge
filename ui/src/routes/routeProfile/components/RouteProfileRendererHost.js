import React from 'react'
import routeProfileRenderer from '../js/routeProfileRenderer'

export default class RouteProfileRendererHost extends React.Component {
    componentDidMount() {
        this.webGlRenderer = routeProfileRenderer(this.renderContainer, this.props.activity.route)
    }
    
    componentWillUnmount() {
        if (this.webGlRenderer) {
            this.webGlRenderer.detach()
            this.webGlRenderer = null
        }
    }

    shouldComponentUpdate(nextProps) {
        return nextProps.activity !== null && this.props.activity === null
    }   

    render() {
        return <div ref={renderContainer => this.renderContainer = renderContainer }></div>
    }
}

RouteProfileRendererHost.propTypes = {
    activity: React.PropTypes.object
}
