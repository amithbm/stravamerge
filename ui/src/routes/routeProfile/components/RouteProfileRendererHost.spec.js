import React from 'react'
import {shallow} from 'enzyme'
import RouteProfileRendererHost from './RouteProfileRendererHost'
import toJson from 'enzyme-to-json'

describe('components', () => {
    describe('<RouteProfileRendererHost />', () => {
        it('renders as expected', () => {
            const component =  shallow(<RouteProfileRendererHost activity={{ activityId: 1 }} />)
            const tree = toJson(component)
            expect(tree).toMatchSnapshot()
        })
    })
})


