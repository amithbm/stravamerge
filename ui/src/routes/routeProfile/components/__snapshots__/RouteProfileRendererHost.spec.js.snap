exports[`components <RouteProfileRendererHost /> renders as expected 1`] = `<div />`;
